examples/src/bin/flash_read_write.rs

```diff
@@ -11,22 +10,57 @@
 
 use embedded_storage::{ReadStorage, Storage};
 use esp_backtrace as _;
+use esp_bootloader_esp_idf::partitions;
 use esp_hal::main;
 use esp_println::println;
 use esp_storage::FlashStorage;
 
+esp_bootloader_esp_idf::esp_app_desc!();
+
 #[main]
 fn main() -> ! {
-    let mut bytes = [0u8; 32];
+    let _ = esp_hal::init(esp_hal::Config::default());
 
     let mut flash = FlashStorage::new();
-
-    let flash_addr = 0x9000;
     println!("Flash size = {}", flash.capacity());
+
+    let mut pt_mem = [0u8; partitions::PARTITION_TABLE_MAX_LEN];
+    let pt = partitions::read_partition_table(&mut flash, &mut pt_mem).unwrap();
+
+    for i in 0..pt.len() {
+        let raw = pt.get_partition(i).unwrap();
+        println!("{:?}", raw);
+    }
+    println!();
+
+    // The app descriptor (if present) is contained in the first 256 bytes
+    // of an app image, right after the image header (24 bytes) and the first section header (8 bytes)
+    let mut app_desc = [0u8; 256];
+    pt.find_partition(partitions::PartitionType::App(
+        partitions::AppPartitionSubType::Factory,
+    ))
+    .unwrap()
+    .unwrap()
+    .as_embedded_storage(&mut flash)
+    .read(32, &mut app_desc)
+    .unwrap();
+    println!("App descriptor dump {:02x?}", app_desc);
+    println!();
+
+    let nvs = pt
+        .find_partition(partitions::PartitionType::Data(
+            partitions::DataPartitionSubType::Nvs,
+        ))
+        .unwrap()
+        .unwrap();
+    let mut nvs_partition = nvs.as_embedded_storage(&mut flash);
+
+    let mut bytes = [0u8; 32];
+    println!("NVS partition size = {}", nvs_partition.capacity());
     println!();
 
-    flash.read(flash_addr, &mut bytes).unwrap();
-    println!("Read from {:x}:  {:02x?}", flash_addr, &bytes[..32]);
+    nvs_partition.read(0, &mut bytes).unwrap();
+    println!("Read from {:x}:  {:02x?}", 0, &bytes[..32]);
```

Comment:
Is the 0 still useful? Could it be `nvs_partition.offset()` or something similar, just so it's still useful info.

Reply:
mhhh - a hardcoded 0 probably not but will change sth here